File: cpp/연속된_수의_합/연속된_수의_합.cpp
#include <string>
#include <vector>

using namespace std;

// 1. 1부터 num - 1 번 만큼 1씩 커지는 등차수열의 형식으로 더해간다.
// 2. 더한 값을 total에서 뺀다.
// 3. 2번에서 나온 값을 num으로 나누면 배열의 0번 원소가 나온다.
// 4. 배열의 0번 원소로부터 1씩 커지는 등차수열로 num 번 까지 값을 answer에 넣는다.

vector<int> solution(int num, int total) {
    vector<int> answer;
    int sum = 0;
    
    for (int i = 1; i < num; ++i) {
        sum += i; // 위 1번의 값을 구해준다.
    }
    
    int start = (total - sum) / num; // 배열의 0번 원소 구하기
    
    for (int i = 0; i < num; ++i) {
        answer.push_back(start + i); // 배열의 0번 원소부터 n개만큼의 원소 answer에 추가
    }
    
    return answer;
}

File: go/연속된_수의_합/연속된_수의_합.go
func solution(num int, total int) []int {
    var result []int
    sum := 0
    // x + (x + 1) + (x + 2) ... = total이다.
    // total에서 자연수만 빼면 num * x가 나온다.
    // total에서 자연수를 뺀 값을 num으로 나누면 x가 나오겠지? -> 그게 초기값
    for i := 1; i < num; i++ {
        sum += i
    }
    // 정답의 0번째 원소를 구한다.
    start := (total - sum) / num
    // append 함수를 사용해서 배열을 채워준다.
    for i := 0; i < num; i++ {
        result = append(result, start + i)
    }
    return result
}